Added fireservicerota documentation

cyberjunky opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

This PR adds documentation for the new fireservicerota integration.

I was forced to close the old PR https://github.com/home-assistant/home-assistant.io/pull/14568  because I want to contribute to other issues, and open PR's for current branch again, and I used 'current' as fork branch preventing this.
Sorry for the inconvenience.

It now also points to 'next' branch instead of 'current'.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/38206
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

cyberjunky committed:
Added fireservicerota documentation

cyberjunky committed:
Correct textlint errors

klaasnicolaas committed:
:ambulance: Fix extra blank lines

klaasnicolaas committed:
:ambulance: Fix textlint error

cyberjunky committed:
Update source/_integrations/fireservicerota.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

cyberjunky committed:
Review: fixed image, remove attributes, third-party, and template sensor

cyberjunky committed:
Updated screenshot

cyberjunky commented:
Created a mess, I will try to clean up ones I know how ❓

cyberjunky committed:
Changed ha_release

cyberjunky committed:
Sigh, another release missed ;-)

balloob commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+  - Sensor
+  - Switch
+ha_iot_class: Cloud Polling
+ha_release: 0.118
```
```suggestion
ha_release: 0.119
```

balloob committed:
Update source/_integrations/fireservicerota.markdown

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+FireServiceRota is a powerful and flexible availability, scheduling and dispatching system for firefighters.
+It's the international brand of the Dutch [BrandweerRooster](https://www.brandweerrooster.nl), which is in use by more than 200 fire stations in The Netherlands.
+
+The `fireservicerota` integration provides you realtime information about incidents (emergency calls) from your local fire station, and the ability to send a response depending on your duty schedule.
```
```suggestion
The FireServiceRota integration provides you real-time information about incidents (emergency calls) from your local fire station and the ability to send a response depending on your duty schedule.
```

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+
+* Sensor: Incoming emergency calls. Metadata contains _among other data_ the location of the incident and a text-to-speech URL. The integration uses a WebSocket client connection with the service to ensure a minimum delay.
+* Binary Sensor: Your current duty status (as scheduled via the FireServiceRota mobile app and/or website)
+* Switch: Enabled for 30 minutes after any emergency call. ‘on’ represents a confirmed response. Use this to automate your emergency call response and save valuable seconds.
```
```suggestion
- Sensor: Incoming emergency calls. Metadata contains _among other data_ the location of the incident and a text-to-speech URL. The integration uses a WebSocket client connection with the service to ensure a minimum delay.
- Binary Sensor: Your current duty status (as scheduled via the FireServiceRota mobile app and/or website).
- Switch: Enabled for 30 minutes after an emergency call. ‘on’ represents a confirmed response. Use this to automate your emergency call response and save valuable seconds.
```

frenck commented on source/_integrations/fireservicerota.markdown:
````diff
@@ -0,0 +1,220 @@
+
+The FireServiceRota integration will log additional information about websocket incidents received, response and duty status gathered, and other messages when the log level is set to `debug`. Add the the relevant lines below to the `configuration.yaml`:
+
+```yaml
````
This code block is incomplete / not closed?

cyberjunky replied:
Fixed

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+
+## Debugging
+
+The FireServiceRota integration will log additional information about websocket incidents received, response and duty status gathered, and other messages when the log level is set to `debug`. Add the the relevant lines below to the `configuration.yaml`:
```
```suggestion
The FireServiceRota integration will log additional information about WebSocket incidents received, response and duty status gathered, and other messages when the log level is set to `debug`. Add the relevant lines below to the `configuration.yaml`:
```

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+<img src='/images/integrations/fireservicerota/dashboard.png' alt='Example of a FireServiceRota dashboard' class="no-shadow"/> 
+
+This screenshot shows what a FireServiceRota dashboard can look like.
+
```
```suggestion
```

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+
+The following entities are created:
+
+### sensor.incidents
```
These entity ID's don't have to match (e.g., renamed, or changed by the end-user). Therefore a descriptive name might be more fitting.

cyberjunky replied:
Good point, changed.

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+
+| Value | Description |
+| ----- | ----------- |
+| `value` | Incident text message.|
```
This is like a bit of a weird table. Maybe it makes more sense to describe the sensor value in normal text in this case.

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+| `latitude` | Latitude.|
+| `longitude` | Longitude.|
+| `address_type` | Type of address, e.g. 'home'.|
+| `formatted_address` | Address in string format.|
```
```suggestion
| Attribute | Description |
| --------- | ----------- |
| `trigger` | Type of trigger, `new` or `update`.|
| `state` | The state of the incident. |
| `created_at` | Date and time when incident was created.|
| `message_to_speech_url` | The URL of the mp3 file containing the spoken text of the incident.|
| `prio` | Priority of the incident, `a1`, `a2`, `b1` or `b2`.|
| `type` | Type of incident, e.g., `incident_alert`.|
| `responder_mode` | Modes of response, e.g., `available_in_schedule_is_acknowledgment`.|
| `can_respond_until` | Date and time until response is accepted.|
| `latitude` | Latitude.|
| `longitude` | Longitude.|
| `address_type` | Type of address, e.g., `home`.|
| `formatted_address` | Address in string format.|

```

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+| `longitude` | Longitude.|
+| `address_type` | Type of address, e.g. 'home'.|
+| `formatted_address` | Address in string format.|
+### binary_sensor.duty
```
Same as above; this doesn't have to match, recommended to use a descriptive name.

cyberjunky replied:
Fixed

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+
+| Value | Description |
+| ----- | ----------- |
+| `value` | 'On' = on duty. 'Off' = no duty scheduled.|
```
Same as above.

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+| `assigned_function_ids` | Function id's, e.g. '540'.|
+| `skill_ids` | Skill id's, e.g. '6, 8'.|
+| `type` | Type, e.g. 'standby_duty'.|
+| `assigned function` | Assigned function, e.g. 'Chauffeur'.|
```
```suggestion
| Attribute | Description |
| --------- | ----------- |
| `start_time` | Start date and time of duty schedule.|
| `end_time` | End date and time of duty schedule.|
| `available` | `true` or `false`.|
| `active` | `true` or `false`.|
| `assigned_function_ids` | Function ID's, e.g., `540`.|
| `skill_ids` | Skill ID's, e.g., `6, 8`.|
| `type` | Type, e.g., `standby_duty`.|
| `assigned function` | Assigned function, e.g., `Chauffeur`.|
```

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+| `type` | Type, e.g. 'standby_duty'.|
+| `assigned function` | Assigned function, e.g. 'Chauffeur'.|
+
+### switch.incident_response
```
Same as above

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+
+| Value | Description |
+| ----- | ----------- |
+| `value` | 'On' = response acknowledged. 'Off' = response rejected.|
```
Same as the above value sections

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+| `arrived_at_station` | 'true' or 'false'.|
+| `available_at_incident_creation` | 'true' or 'false'.|
+| `active_duty_function_ids` | Active function id's, e.g. '540'.|
+
```
```suggestion
| Attribute | Description |
| --------- | ----------- |
| `user_name` | Your username.|
| `assigned_skill_ids` | Assigned skill ID's.|
| `responded_at` | Time responded.|
| `start_time` | Incident repsonse start time.|
| `status` | Status of repsonse, e.g., `pending`.|
| `reported_status` | Reported status, e.g., `shown_up`.|
| `arrived_at_station` | `true` or `false`.|
| `available_at_incident_creation` | `true` or `false`.|
| `active_duty_function_ids` | Active function ID's, e.g., `540`.|
```

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+With Automation you can configure one or more of the following useful actions:
+
+1. Sound an alarm and/or switch on lights when an emergency incident is received.
+1. Use text to speech to play incident details via a mediaplayer while getting dressed.
```
```suggestion
1. Use text to speech to play incident details via a media player while getting dressed.
```

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+
+1. Sound an alarm and/or switch on lights when an emergency incident is received.
+1. Use text to speech to play incident details via a mediaplayer while getting dressed.
+1. Respond with an response acknowledgement using a door-sensor when leaving the house or by pressing a button to let your teammates know you are underway.
```
```suggestion
1. Respond with a response acknowledgment using a door-sensor when leaving the house or by pressing a button to let your teammates know you are underway.
```

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,220 @@
+      platform: state
+      entity_id: sensor.incidents
+      attribute: message_to_speech_url
+    condition: "{{ state_attr('sensor.incidents', 'message_to_speech_url') != 'None' }}"
```
Conditions can work with attributes too!

```yaml
condition: not
conditions:
  condition: state
  entity_id: sensor.incidents
  attribute: message_to_speech_url
  state: "None"
```

cyberjunky replied:
Nice, fixed.

frenck commented:
Thanks for the PR @cyberjunky, I've left some review comments. Could you please take a look? Thanks 👍

cyberjunky committed:
Better writing

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

cyberjunky committed:
Corrected list

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

cyberjunky committed:
Spelling correction

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

cyberjunky committed:
Remove blank line

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

cyberjunky committed:
Spelling correction

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

cyberjunky committed:
Spelling correction

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

cyberjunky committed:
Fixed quotes

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

cyberjunky committed:
Closed code block

cyberjunky committed:
Use descriptive names for the entities

cyberjunky committed:
Corrected config flow order to reflect latest code

cyberjunky committed:
Fixed quotes and spelling errors

cyberjunky committed:
Better syntax for condition

cyberjunky commented:
> Thanks for the PR @cyberjunky, I've left some review comments. Could you please take a look? Thanks +1

@frenck I think I have addressed all suggestions.

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,226 @@
+
+### Duty Binary Sensor
+
+| Value | Description |
+| ----- | ----------- |
+| `value` | `on` = on duty. `off` = no duty scheduled.|
```
As mentioned before, I really think we do not need a table here.
It makes more sense to have a written introduction about the sensor and mention the values in that description. (Removing the whole table).

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,226 @@
+
+### Incident Response Switch
+
+| Value | Description |
+| ----- | ----------- |
+| `value` | `on` = response is acknowledged. `off` = the response is rejected.|
```
As mentioned before, I really think we do not need a table here.
It makes more sense to have a written introduction about the sensor and mention the values in that description. (Removing the whole table).

cyberjunky replied:
@frenck I change them, what do you think?

frenck commented on source/_integrations/fireservicerota.markdown:
```diff
@@ -0,0 +1,226 @@
+  - alias: 'Play TTS incident details when incident is received'
+    trigger:
+      platform: state
+      entity_id: sensor.incidents
+      attribute: message_to_speech_url
+    condition: not
+    conditions:
+      condition: state
+      entity_id: sensor.incidents
+      attribute: message_to_speech_url
+      state: "None"
+    action:
+      - service: media_player.play_media
+        data_template:
```
You might want to test this, it seems invalid now.

cyberjunky committed:
Fix automation example

cyberjunky committed:
Replace the tables with description texts

cyberjunky committed:
Fixed URL lint

cyberjunky committed:
Fixed spelling

cyberjunky committed:
Spelling fixes

frenck commented:
@cyberjunky I kinda meant just the first tables with the single value 😓 (the state, not the attributes)

cyberjunky commented:
> @cyberjunky I kinda meant just the first tables with the single value sweat (the state, not the attributes)

Haha. LOL oke, wil revert and adapt the value only.

cyberjunky committed:
Reverted to tables for attributes only

frenck reviewed (approved):
Thanks, @cyberjunky! 👍
